components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -1671,18 +1715,18 @@ AuthenticatedUser getAppOwner(OAuthConsumerAppDTO application,
         if (StringUtils.isNotBlank(applicationOwnerInRequest)) {
             String tenantAwareAppOwnerInRequest = MultitenantUtils.getTenantAwareUsername(applicationOwnerInRequest);
             try {
-                if (CarbonContext.getThreadLocalCarbonContext().getUserRealm().
-                        getUserStoreManager().isExistingUser(tenantAwareAppOwnerInRequest)) {
-                    // Since the app owner sent in OAuthConsumerAppDTO is a valid one we set the appOwner to be
-                    // the one sent in the OAuthConsumerAppDTO.
-                    String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
-                    appOwner = buildAuthenticatedUser(tenantAwareAppOwnerInRequest, tenantDomain);
+                // Since the app owner sent in OAuthConsumerAppDTO is a valid one we set the appOwner to be
+                // the one sent in the OAuthConsumerAppDTO.
+                String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+                Optional<User> maybeAppOwner = OAuthUtil.getUser(tenantDomain, tenantAwareAppOwnerInRequest);
+                if (maybeAppOwner.isPresent()) {
+                    appOwner = new AuthenticatedUser(maybeAppOwner.get());
                 } else {
                     LOG.warn("OAuth application owner user name " + applicationOwnerInRequest +
                             " does not exist in the user store. Using user: " +
                             defaultAppOwner.toFullQualifiedUsername() + " as app owner.");
                 }
-            } catch (UserStoreException e) {
+            } catch (IdentityApplicationManagementException e) {
                 throw handleError("Error while retrieving the user store manager for user: " +
```

Comment:
is this the correct error message now that the exception has been changed?

Reply:
relsolved